backend/src/main/java/ca/mcgill/purposeful/controller/IdeaController.java

```diff
@@ -93,4 +97,19 @@ public IdeaDTO modifyIdea(
             iconUrlId);
     return new IdeaDTO(modifiedIdea);
   }
+
+  /**
+   * Remove an idea by its id
+   *
+   * @param id the idea's id
+   * @return a response entity with a message instance and the HttpStatus
+   */
+  @DeleteMapping({"/{id}", "/{id}/"})
+  @PreAuthorize("hasAnyAuthority('User', 'Moderator', 'Owner')")
+  public ResponseEntity<String> removeIdea(@PathVariable String id) {
+    // call service layer
```

Comment:
We should add an authentication so that not anyone can call this endpoint and delete the idea of others

Reply:
Great point. Can someone outside the system get access to the idea id?

Reply:
I mean someone can always try and guess a uuid

But even if you are logged in, user2 should not be able to delete the idea of user1

Reply:
Absolutely. Entering the idea_id and user_id as path variables then checking their match in the database may solve this issue

Reply:
Is it not possible to get the user from the authentication?

Reply:
I am unsure about that

Reply:
Maybe we can use this : https://www.baeldung.com/get-user-in-spring-security

Reply:
Awesome, looks like it's exactly what we want! Due to time constraints, I will add it to the known bugs list and take a look at it during spring B.

Reply:
That sounds good

Reply:
Authentification added in T076